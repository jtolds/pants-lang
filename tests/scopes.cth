x = (3)
print x
{ |x| print x; x := 4; print x }(2)
print x

print.

x = 1
y = { print x }
y.
print x
x := 2
y.
print x

print.

x = 1
y = { print x }
y.
print x
x = 2
y.
print x

print.

z = {
  (
    cont 3
  )
  cont 4
}

print z()

z = {
  cont 4
  (
    cont 3
  )
}

print z()

print.

x = 10
(
  x = 11
)
print x

x = 10
(
  x := 11
)
print x
