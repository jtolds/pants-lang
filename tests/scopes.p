# PANTS OPTIONS: --skip-prelude

x = (3)
println x
{ |x| println x; x := 4; println x }(2)
println x

println.

x = 1
y = { println x }
y.
println x
x := 2
y.
println x

println.

x = 1
y = { println x }
y.
println x
x = 2
y.
println x

println.

x = 1
y = ( z = x; { println z } )
y.
println x
x = 2
y.
println x

println.

z = {
  (
    cont 3
  )
  cont 4
}

println z()

z = {
  cont 4
  (
    cont 3
  )
}

println z()

println.

x = 10
(
  x = 11
)
println x

x = 10
(
  x := 11
)
println x

println.

var = 100
get = { println var }
set = { |val| var := val }
var = 101
println var
get.
set 102
println var
get.

println.

var2 = 200
get = { println var2 }
set = { |val| var2 := val }
var2 := 201
println var2
get.
set 202
println var2
get.

println.

recursion = {|x|
  if (< x 10) {
    recursion (+ x 1)
  }
  println x
}
recursion 1

println.

recursion = {|x, y|
  x := y
  println x
  if (< y 10) {
    recursion 1 (+ y 1)
  }
  println x
}
recursion 1 1

println.

x = 3
test = {
  println x
  x = 4
  println x
}
test.
test.

println.
