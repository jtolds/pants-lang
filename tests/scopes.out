3
2
4
3

1
1
2
2

1
1
1
2

3
4

10
11

return code: 0
